### PR TITLE
Fix pcc drop in Bloom, Codegen, Gptneo pytorch model

### PR DESCRIPTION
**Pull request by kamalrajkannan78**
### Ticket

- N/A

### Problem description

- A  PCC drop was observed in Bloom, Codegen, Gptneo models due to inplace update operation

### What's changed

- The problematic in-place update in the attention mask update logic has been monkey patched to avoid in-place modifications.

### Logs

- [apr28_bloom_before_fix.log](https://github.com/user-attachments/files/19941800/apr28_bloom.log)
- [apr28_bloom_after_fix.log](https://github.com/user-attachments/files/19941799/apr28_bloom_after_fix.log)
- [apr28_codegen_before_fix.log](https://github.com/user-attachments/files/19941817/apr28_codegen_before_fix.log)
- [apr28_codegen_after_fix.log](https://github.com/user-attachments/files/19941801/apr28_codegen_after_fix.log)
- [apr28_gpt_neo_before_fix.log](https://github.com/user-attachments/files/19941823/apr28_gpt_neo.log)
- [apr28_gpt_neo_after_fix.log](https://github.com/user-attachments/files/19941818/apr28_gpt_neo_after_fix.log)

### CODE REVIEW

File: forge/test/models/pytorch/text/gptneo/test_gptneo.py [hunk 1]
```diff
@@ -16,6 +17,54 @@
 
 from test.utils import download_model
 
+
```
Comment (saiarthiraguram) [inline forge/test/models/pytorch/text/gptneo/test_gptneo.py:20]: if all the attentions patches are same , move it to model utils, and import them for these models

**Review by saiarthiraguram — APPROVED** [review]

File: forge/test/models/pytorch/text/phi2/test_phi2.py [hunk 2]
```diff
@@ -16,50 +15,9 @@
...
-                part_1 = part_1.masked_fill(padding_mask, min_dtype)
-                causal_mask = torch.cat([part_1, part_2], dim=-1)
-            else:
-                causal_mask = causal_mask.masked_fill(padding_mask, min_dtype)
-
-    return causal_mask
-
+from test.models.models_utils import (
+    _prepare_4d_causal_attention_mask_with_cache_position,
+)
```
Comment (vkovinicTT) [inline forge/test/models/pytorch/text/phi2/test_phi2.py:20]: Did you check whether the model works since this change (I know it's trivial but please check just in case :)

**Review by vkovinicTT — APPROVED** [review]
I am approving this, but can you please send me the link to this original function that you are changing :))

**Comment by kamalrajkannan78** [comment]
> I am approving this, but can you please send me the link to this original function that you are changing :))

-  Bloom -https://github.com/huggingface/transformers/blob/5d7739f15a6e50de416977fe2cc9cb516d67edda/src/transformers/models/bloom/modeling_bloom.py#L801C5-L854C27
- codegen - https://github.com/huggingface/transformers/blob/5d7739f15a6e50de416977fe2cc9cb516d67edda/src/transformers/models/codegen/modeling_codegen.py#L644C5-L698C1
- GPTNeo - https://github.com/huggingface/transformers/blob/5d7739f15a6e50de416977fe2cc9cb516d67edda/src/transformers/models/gpt_neo/modeling_gpt_neo.py#L858C5-L911C27
- Phi2 - https://github.com/huggingface/transformers/blob/5d7739f15a6e50de416977fe2cc9cb516d67edda/src/transformers/models/phi/modeling_phi.py#L1086C5-L1139C27

File: forge/test/models/pytorch/text/phi2/test_phi2.py (hunk 2 above)
Comment (kamalrajkannan78) [reply forge/test/models/pytorch/text/phi2/test_phi2.py:20]: I checked it now. It's working.
